packages/components/Shimmer/ios/ShimmerContainerView.swift

```diff
@@ -0,0 +1,144 @@
+import Foundation
+import FluentUI
+
+
+@objc(MSFShimmerContainerView)
+class ShimmerContainerView: UIView {
+
+	@objc var appearance: NSDictionary = NSDictionary() {
+		didSet {
+			updateShimmerViewAppearance()
+		}
+	}
+	
+	@objc var shimmerAppearance: NSDictionary = NSDictionary() {
+		didSet {
+			updateShimmerAppearance()
+		}
+	}
+
+// MARK: - UIView
+
+	override public func layoutSubviews() {
+		super.layoutSubviews()
+
+		// Shimmer, by default, only considers top-level views when shimmering. This is
+		// incompatible with Flexbox, which wants you to wrap everything in more views to achieve
+		// a specific layout. The solution is to get a list of all the leaves of the subview tree
+		// and shimmer those by creating "mirror" views to mirror the RCTViews in a flat heirarchy in
+		// the shimmer container.
+		mirrors.forEach { $0.removeFromSuperview() }
+		mirrors = []
+
+		var leaves = [UIView]()
+		containerLeaves(view: self, leaves: &leaves)
```

Comment:
nit; can we find a better name for this recursive function? searchLeaves(in view:UIView, output: inout [UIview]) ?

Reply:
Used that name in FUA